asapdiscovery-ml/asapdiscovery/ml/utils.py

```diff
@@ -876,7 +878,54 @@ def load_exp_data(fn, achiral=False, return_compounds=False):
         return exp_dict
 
 
-def load_weights(model, wts_fn):
+def check_model_compatibility(model, to_load, check_weights=False):
+    """
+    Checks if a PyTorch file or state_dict is compatible with a model.
+
+    Parameters
+    ----------
+    model : torch.nn.Module
+        The model to check.
+    to_load : Union[str, Path, Dict[str, torch.Tensor]]
+        The path to the PyTorch file, or a dictionary of the state dict.
+    check_weights : bool, default=False
+        Whether to check the weights of the model and the PyTorch file.
+
+    Returns
+    -------
+    None
+
+    """
+    # Load the PyTorch file
+    if isinstance(to_load, str) or isinstance(to_load, Path):
+        test_state_dict = torch.load(to_load, map_location=torch.device("cpu"))
+
+    elif isinstance(to_load, dict):
+        test_state_dict = to_load
+
+    # Get the state dicts of the model and the PyTorch file
+    model_state_dict = model.state_dict()
+
+    # Check the model architecture
+    if set(model_state_dict.keys()) != set(test_state_dict.keys()):
+        raise ValueError("Model architecture doesn't match.")
+
+    # Check the model weights
+    if check_weights:
+        for key in model_state_dict.keys():
+            if model_state_dict[key].shape != test_state_dict[key].shape:
+                raise ValueError("Model weights shape doesn't match the file.")
+
+            if not torch.allclose(
+                model_state_dict[key], test_state_dict[key], atol=1e-4
+            ):
+                raise ValueError("Model weights don't match the file.")
```

Comment:
is there a scenario where we would want to check that a model has the same weights as a file?

Reply:
I guess possibly if comparing that two state dicts are equal, but you are right probably not needed right now.